cas-server-support-osf/src/test/java/io/cos/cas/authentication/handler/support/OpenScienceFrameworkPrincipalFromRequestRemoteUserNonInteractiveCredentialsActionTests.java

```diff
@@ -39,15 +42,53 @@
 
     private static final String CONST_ORCID_CLIENT_NAME = "OrcidClient";
 
+    @Test (expected = RemoteUserFailedLoginException.class)
+    public void handleInstitutionMissingUsername() throws Exception {
+        final MockHttpServletRequest mockHttpServletRequest = AbstractTestUtils.getRequestWithShibbolethHeaders();
+        final MockRequestContext mockContext = AbstractTestUtils.getContextWithCredentials(mockHttpServletRequest);
+        final CentralAuthenticationService centralAuthenticationService = mock(CentralAuthenticationService.class);
+        final MockNormalizeRemotePrincipal osfRemoteAuthenticate
+                = new MockNormalizeRemotePrincipal(centralAuthenticationService);
+
+        final OpenScienceFrameworkCredential osfCredential = new OpenScienceFrameworkCredential();
+        osfCredential.setUsername("");
+        osfCredential.setInstitutionId(AbstractTestUtils.CONST_INSTITUTION_ID);
+        try {
+            osfRemoteAuthenticate.notifyRemotePrincipalAuthenticated(osfCredential);
```

Comment:
`osfRemoteAuthenticate.doExcute()`fails to work here since the one exception we want to verify is abstracted and handled by the flow.